src-ghc/Pact/Types/ECC.hs

```diff
@@ -0,0 +1,62 @@
+{-# LANGUAGE PackageImports #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Pact.Types.ECC where
+
+import Crypto.PubKey.ECC.Generate (generate)
+import Crypto.PubKey.ECC.Types (Point(..),getCurveByName,CurveName(..))
+import Crypto.PubKey.ECC.ECDSA (PublicKey(..),PrivateKey(..),Signature(..),signWith,verify)
+import Crypto.Hash (hash,Digest,Keccak_256(..),SHA3_256(..))
+import qualified Data.ByteArray as BA
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Base16 as B16
+import Data.ByteString (ByteString)
+import Pact.Types.Util (toB16Text)
+import Data.Text (Text)
+import Data.Monoid ((<>))
+
+
+gen :: IO (PublicKey,PrivateKey)
+gen = generate (getCurveByName SEC_p256k1)
+
+formatEthPublicKey :: PublicKey -> ByteString
```

Comment:
Some questions:

- PublicKeys includes a `public_q :: PublicPoint`, which can be two integers. Are these two integers the public key? Or do you have to use the points and the curve information to get the public key? I ask because in the example [here](https://kobl.one/blog/create-full-ethereum-keypair-and-address/), the public key seems to be just one hexadecimal.
- Following up on the previous question, I see that you convert each point to hex and then concatenate them. Is this correct?

Reply:
The concatenating is based on the old Haskell code I found here, no idea if it's accurate: https://github.com/jamshidh/ethereum-data-leveldb/blob/master/src/Blockchain/Data/Address.hs#L38

Regarding your example, command-line tools just represent these things as hex (which maybe under the hood do the same concat?)